wasp-ai/src/client/pages/ResultPage.jsx

```diff
@@ -73,10 +74,16 @@ export const ResultPage = () => {
   }, [appId]);
 
   useEffect(() => {
-    if (currentStatus.status === "success") {
-      setIsStarRepoOpen(true);
+    const currentAppStatus = appGenerationResult?.project?.status;
+    if (currentAppStatus === "in-progress") {
+      setIsRunning(true)
+    } else if (currentAppStatus === "success" && isRunning) {
+      setIsSuccessModalOpen(true)
+      setIsRunning(false)
+    } else if (isRunning) {
+      setIsRunning(false)
     }
-  }, [currentStatus])
+  }, [appGenerationResult?.project?.status])
```

Comment:
While this probably does the trick, it feels a bit peculiar when reading it -> why are we setting is running together with modal? Why do we do it at all, when we have `appGenerationResult.project.status`?
We know the answer, but it is not obvious and this knowledge will get lost.

To be clear, this whole file is quite messy and not written with clean code (we were in hurry so we let it slide), but we have a bit of time now so it is good opportunity to at least try to get these new modifications as clean as possible.

Quick google search tells me that best thing to track and obtain old state is via useRef: https://stackoverflow.com/questions/53446020/how-to-compare-oldvalues-and-newvalues-on-react-hooks-useeffect -> so you introduce custom hook `usePrevious` it seems. Maybe that is the best way to go?
What you could possibly do then is have `prevGenerationResultStatus`, and then in `useEffect` check that to decide what you do with `setIsSuccessModalOpen`.

So instead of tracking `isRunning`, we track `prevGenerationResultStatus`. One might wonder why we track that, but they can easily figure that out, and they don't have to wonder what is correct or more correct: `isRunning` vs `appGeneration.project.status === 'running'`, and if they are in sync.
Also, our logic for dealing with `setIsSuccessModalOpen` will be much shorter and standalone, decoupled from logic of updating `prevGenerationResultStatus`.